tests/api-controller-acceptance-tests/apicontroller/main_test.go

```diff
@@ -85,8 +86,8 @@ func loadKubeConfigOrDie() *rest.Config {
 
 func testWithNamespace(m *testing.M) int {
 	catchInterrupt()
-	createNamespace()
 	defer deleteNamespace()
+	createNamespace()
```

Comment:
`createNamespace` function is invoking `log.Fatalf()` which logs an error message and invokes ox.Exit(1) under the hood.  `os.Exit` work in given way : 

> The program terminates immediately; deferred functions are not run.

So this change (PR) does not solve the root cause. To solve the issue `createNamespace` should *not* invoke `log.Fatalf()` but `log.Errorf()` inestad and this error should be  handled in the main